SigAverager/SigAverager/signal_averager_decorator.py

```diff
@@ -0,0 +1,89 @@
+"""Signal Averaging."""
+import numpy as np
+import SigAverager.cwg
+
+Number_of_iterations = 2048
+
+
+def n_iterations(n):
+    """Generate Carrier Wave with AWGN.
+
+    Parameters
+    ----------
+    n: int
+        n is the number of averages to perform
+    """
+
+    def inner(f):
+        def wrapper(*args, **kwargs):
+            rv = np.zeros(args[4])
+            for i in range(n):
+                print(f"Running {f.__name__}:{i}")
+                rv = f(rv, *args, **kwargs)
```

Comment:
I get the following output from pytest:

```
        # Compute Averaged Signal
>       averaged_signal = SigAverager.signal_averager(
            cw_scale, cw_freq, sampling_frequency, noise_scale, num_samples, n_iter
        )

testing/test_dsp_signal_averager.py:45:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (0.01, 75000000.0, 1712000000.0, 0.4, 8192, 2048), kwargs = {}, rv = array([0., 0., 0., ..., 0., 0., 0.]), i = 0

    def wrapper(*args, **kwargs):
        rv = np.zeros(args[4])
        for i in range(n):
            print(f"Running {f.__name__}:{i}")
>           rv = f(rv, *args, **kwargs)
E           TypeError: signal_averager() takes 6 positional arguments but 7 were given

../../.venv/lib/python3.8/site-packages/SigAverager/signal_averager_decorator.py:22: TypeError
```

So what's happening here is that you're adding an extra argument in the front that the wrapped function isn't expecting.

I think you could probably get it to work, if you lopped the first one off `args`, in the following manner:
```suggestion
                rv = f(rv, *args[1:], **kwargs)
```